app/src/js/server/main.ts

```diff
@@ -91,6 +92,34 @@ async function makeBotManager (
   }
 }
 
+/**
+ * Launch the redis server
+ */
+async function launchRedisServer (config: ServerConfig) {
+  let redisDir = './'
+  if (config.database === 'local') {
+    redisDir = config.data
+  }
+
+  const redisProc = child.spawn('redis-server', [
+    getRedisConf(),
+    '--port', `${config.redisPort}`,
+    '--bind', '127.0.0.1',
+    '--dir', redisDir,
+    '--protected-mode', 'yes']
+  )
+  redisProc.stdout.on('data', (data) => {
+    process.stdout.write(data)
+  })
+
+  redisProc.stderr.on('data', (data) => {
+    process.stdout.write(data)
+  })
+
+  // wait 1 s before trying to connect to the server
+  await sleep(1000)
```

Comment:
Is there any other signal we can wait for?

Reply:
I can't find any signal Redis emits when it starts up. One alternative would be to wait a smaller time (like 100 ms) in a loop, try to connect, and catch the failure. So it tries to connect every 100 ms. We could also have some maximum cutoff time (like 2-3 s).

Reply:
Long polling sounds better.

Reply:
Ok, so this means leave it at 1 second?

Reply:
I meant taking the alternative approach you mentioned. Wait for a small amount of time, and check the status. Raise failure after a timeout.

Reply:
It turns out Redis has polling built in: https://github.com/NodeRedis/node-redis/issues/138
So I just removed the sleep, and disabled the "connection failed" error logging by default